### DISCUSSION TITLE
MICCAI surgtoolloc preprocess_detect_scene_and_split_fold missing train.csv?

### DISCUSSION THREAD

**Discussion by lukigg** [Project-MONAI/tutorials#1439] [Q&A] [answered]
Hello, 

I am working currently on this notebook https://github.com/Project-MONAI/tutorials/blob/main/competitions/MICCAI/surgtoolloc/preprocess_detect_scene_and_split_fold.ipynb.

I load labels.csv as df and extracted images from preprocess_extract_images_from_video notebook as img_dir. But then at the point Splitting folds based on scene number using iterative stratification in the linked notebook above the authors use 
load df = pd.read_csv("../train.csv") 
which I am unsure of where should I get as it's not in the MICCAI Challange dataset and on GitHub 🤔.

**Comment by KumoLiu** [comment]
Hi @yiheng-wang-nv, could you please take a look at this question?
Thanks in advance!

**Reply by lukigg** [reply]
Thank you @KumoLiu again!

**Comment by yiheng-wang-nv** [comment, marked as answer]
Hi @lukigg , sorry for misleading. 
The `train.csv` is actually produced in: `Load data` section, from:
```
df = pd.read_csv('/raid/surg/_release/training_data/labels.csv')[['clip_name', 'tools_present']]
img_dir = '/raid/surg/image640_blur/'
cpu_ct = 32


def split_label(s):
    return [x.strip(' ') for x in s[1:-1].split(',')]


label_lst = df.tools_present.apply(split_label).values.tolist()
label_lst = [x for xs in label_lst for x in xs]

labels = pd.Series(label_lst).value_counts().index.values[1:]
for lb in labels:
    df[lb] = df.tools_present.str.count(lb)

labels = df.columns.values[2:]
df.shape
```

 but the save to csv line is not existing.

I will submit a PR to fix this issue. Thanks for posting this issue!

**Reply by yiheng-wang-nv** [reply]
PR is:
https://github.com/Project-MONAI/tutorials/pull/1441/files
It would be merged soon. Thanks again @lukigg

**Reply by lukigg** [reply]
It worked now, I will continue working on it thank you for your support!
